src/odemis/driver/xt_client.py

```diff
@@ -349,20 +468,579 @@ def set_autofocusing(self, channel, state):
+            if beam_shift != self.beamShift.value:
+                self.beamShift._value = beam_shift
+                self.beamShift.notify(beam_shift)
+            rotation = self.parent.get_rotation()
+            if rotation != self.rotation.value:
+                self.rotation._value = rotation
+                self.rotation.notify(rotation)
+            fov = self.parent.get_scanning_size()[0]
+            if fov != self.horizontalFoV.value:
+                self.horizontalFoV._value = fov
+                self.horizontalFoV.notify(fov)
+                mag = self._hfw_nomag / fov
+                self.magnification._value = mag
+                self.magnification.notify(mag)
+        except Exception:
+            logging.exception("Unexpected failure when polling settings")
+
+    def _setDwellTime(self, dwell_time):
+        self.parent.set_dwell_time(dwell_time)
+        return self.parent.get_dwell_time()
+
+    def _setVoltage(self, voltage):
+        self.parent.set_ht_voltage(voltage)
+        return self.parent.get_ht_voltage()
+
+    def _setBlanker(self, blank):
+        """True if the the electron beam should blank, False if it should be unblanked."""
+        if blank:
+            self.parent.blank_beam()
+        else:
+            self.parent.unblank_beam()
+        return self.parent.beam_is_blanked()
+
+    def _setSpotSize(self, spotsize):
+        self.parent.set_ebeam_spotsize(spotsize)
+        return self.parent.get_ebeam_spotsize()
+
+    def _setBeamShift(self, beam_shift):
+        self.parent.set_beam_shift(*beam_shift)
+        return self.parent.get_beam_shift()
+
+    def _setRotation(self, rotation):
+        self.parent.set_rotation(rotation)
+        return self.parent.get_rotation()
+
+    def _setHorizontalFoV(self, fov):
+        self.parent.set_scanning_size(fov)
+        fov = self.parent.get_scanning_size()[0]
+        mag = self._hfw_nomag / fov
+        self.magnification._value = mag
+        self.magnification.notify(mag)
+        self._updateDepthOfField()
+        return fov
+
+    def _updateDepthOfField(self):
+        fov = self.horizontalFoV.value
+        # Formula was determined by experimentation
+        K = 100  # Magical constant that gives a not too bad depth of field
+        dof = K * (fov / 1024)
+        self.depthOfField._set_value(dof, force_write=True)
+
+
+class Stage(model.Actuator):
+    """
+    This is an extension of the model.Actuator class. It provides functions for
+    moving the TFS stage and updating the position.
+    """
+
+    def __init__(self, name, role, parent, rng=None, **kwargs):
+        if rng is None:
+            rng = {}
+        stage_info = parent.stage_info()
+        if "x" not in rng:
+            rng["x"] = stage_info["range"].get("x", (0, 100e-6))
+        if "y" not in rng:
+            rng["y"] = stage_info["range"].get("y", (0, 100e-6))
+        if "z" not in rng:
+            rng["z"] = stage_info["range"].get("z", (0, 100e-6))
```

Comment:
Are there any chance that `stage_info` doesn't contain any of these axes? I'd guess that if there are missing, something is really odd, and it's better to fail, instead of guessing 100µm. => just write like ` rng["x"] = stage_info["range"]["x"]`